crates/interfaces/src/test_utils/generators.rs

```diff
@@ -204,4 +206,49 @@ mod test {
             assert_eq!(recovered, expected);
         }
     }
+
+    #[test]
+    fn test_sign_eip_155() {
+        // reference: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md#example
+        let transaction = Transaction::Legacy(TxLegacy {
+            chain_id: Some(1),
+            nonce: 9,
+            gas_price: 20 * 10_u128.pow(9),
+            gas_limit: 21000,
+            to: TransactionKind::Call(hex!("3535353535353535353535353535353535353535").into()),
+            value: 10_u128.pow(18),
+            input: Bytes::default(),
+        });
+
+        // TODO resolve dependency issue
+        // let mut encoded = BytesMut::new();
+        // transaction.encode(&mut encoded);
+        // let expected =
+        // hex!("ec098504a817c800825208943535353535353535353535353535353535353535880de0b6b3a764000080018080");
+        // assert_eq!(expected, encoded.as_ref());
```

Comment:
What's the issue here? Should we remove this todo?

Reply:
Basically, I'd like to fully follow up the reference example, https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md#example. But for calling `transaction.encode()`, `reth-interfaces` needs `reth-primitives` that causes a cross dependency issue so I left a TODO at here. We already checked the `hash` of the transaction and it could be enough even though it's not follow up fully the reference example. If you think it's already enough, I'll remove this TODO.

Reply:
That's odd, isn't interfaces already importing primitives? What's the cross dependency? If we have a refactor we could do to ease things, also open to it.